packages/loader/src/gltf/parser/GLTFParserContext.ts

```diff
@@ -47,10 +59,71 @@ export class GLTFParserContext {
   }
 
   /**
-   * Get all the buffer data.
+   * Get the resource of the specified type.
+   * @param type - The type of resource
+   * @param index - The index of resource, default all
    */
-  getBuffers(): Promise<ArrayBuffer[]> {
-    return Promise.resolve(this._buffers);
+  get<T>(type: GLTFParserType, index?: number): Promise<T> {
+    const cacheKey = `${type}:${index}`;
+    const cache = this._cache;
+    let promise: Promise<T> = cache.get(cacheKey);
+
+    if (!promise) {
+      if (index !== undefined) {
+        const allPromise: Promise<T> = cache.get(`${type}:${undefined}`);
+        if (allPromise) {
+          return allPromise.then((res) => res[index]);
+        }
+      }
+
+      promise = GLTFParserContext._parsers[type].parse(this, index);
+      cache.set(cacheKey, promise);
+    }
+
+    return promise;
+  }
+
+  /** @internal */
+  _parse(): Promise<GLTFResource> {
+    return this.get<IGLTF>(GLTFParserType.JSON).then((json) => {
+      this.glTF = json;
+
+      return Promise.all([
+        this.get<Texture2D[]>(GLTFParserType.Texture),
+        this.get<Material[]>(GLTFParserType.Material),
+        this.get<ModelMesh[][]>(GLTFParserType.Mesh),
+        this.get<Entity[]>(GLTFParserType.Entity),
+        this.get<Skin[]>(GLTFParserType.Skin),
+        this.get<AnimationClip[]>(GLTFParserType.Animation),
+        this.get<Entity>(GLTFParserType.Scene),
+        this.get<void>(GLTFParserType.Validator)
+      ]).then(([textures, materials, meshes, entities, skins, animations, defaultSceneRoot]) => {
+        const {
+          materialsPromiseInfo,
+          defaultSceneRootPromiseInfo,
+          texturesPromiseInfo,
+          meshesPromiseInfo,
+          animationClipsPromiseInfo
+        } = this;
+
+        const glTFResource = this.glTFResource;
+
+        glTFResource.textures = textures;
+        glTFResource.materials = materials;
+        glTFResource.meshes = meshes;
+        glTFResource.entities = entities;
+        glTFResource.skins = skins;
+        glTFResource.animations = animations;
+
+        texturesPromiseInfo.resolve(textures);
+        materialsPromiseInfo.resolve(materials);
+        meshesPromiseInfo.resolve(meshes);
+        animationClipsPromiseInfo.resolve(animations);
```

Comment:
This will cause circular dependency bug,  sub Promise must resolve when Texture2Ds loaded

Reply:
done